files/telegraf_production.conf

```diff
@@ -0,0 +1,59 @@
+[global_tags]
+[agent]
+  interval = "10s"
+  round_interval = true
+  metric_batch_size = 1000
+  metric_buffer_limit = 10000
+  collection_jitter = "5s"
+  flush_interval = "30s"
+  flush_jitter = "5s"
+  precision = "0s"
+  debug = false
+  quiet = true
+  hostname = "{{.Hostname}}"
+  omit_hostname = false
+[[outputs.influxdb]]
+  urls = ["udp://influxdb:9096"] # required
+  database = "telegraf" # required
+  retention_policy = ""
+  write_consistency = "any"
+  timeout = "5s"
+  udp_payload = "61440B"
+[[inputs.cpu]]
+  percpu = true
+  totalcpu = true
+  collect_cpu_time = false
+  report_active = false
+  core_tags = false
+[[inputs.disk]]
+  ignore_fs = ["tmpfs", "devtmpfs", "devfs", "iso9660", "overlay", "aufs", "squashfs", "nsfs", "fuse.portal", "fuse.gvfsd-fuse", "fuse.lxcfs", "fuse.dockerfs", "fuse", "hugetlbfs", "mqueue", "vfat"]
+[[inputs.diskio]]
+[[inputs.kernel]]
+[[inputs.mem]]
+[[inputs.processes]]
+[[inputs.swap]]
+[[inputs.system]]
+[[inputs.docker]]
+  endpoint = "unix:///var/run/docker.sock"
+  timeout = "5s"
+  perdevice = false
+  perdevice_include = ["cpu", "blkio", "network"]
+[[inputs.influxdb]]
+  urls = [
+    "http://influxdb:8086/debug/vars"
+  ]
+  timeout = "5s"
+[[inputs.internal]]
+  collect_memstats = true
+# [[inputs.linux_sysctl_fs]]
+[[inputs.netstat]]
+# [[inputs.procstat]]
+  # pattern = "."
+  # pid_finder = "native"
+  # tag_with = ["cmdline"]
+[[inputs.wireless]]
+  host_proc = "/hostfs/proc/1"
+# [[inputs.docker_log]]
+#     endpoint = "unix:///var/run/docker.sock"
+[[inputs.redis]]
+  servers = ["tcp://redis-local:6379","tcp://redis-master:6379"]
```

Comment:
@mariana-dias-alves can you check the production configuration populates the dashboards we have in 2.3.1 ?

Reply:
[Screencast from 23-05-2024 11:55:07.webm](https://github.com/MOV-AI/containers-telegraf/assets/139454338/fdb5a056-14da-46ad-9ba1-2f7534674346)

Reply:
are these the dashboards you are referring to?

Reply:
yes, so I see that in the system dashboard and the top dashboard we are loosing data which might be seen as breaking change on 2.3.1

Reply:
for the top dashboard, the query is based on procstat:
`SELECT LAST(pid) as "pid", LAST(cpu_usage) as "cpu_usage", LAST(memory_usage) as "mem_usage", -LAST(cpu_usage) as "reverse_cpu_usage" FROM "telegraf"."autogen"."procstat" WHERE time > now() - 30s GROUP BY cmdline FILL(0.0)`

Reply:
for the system dashboard:
- net cell
`SELECT non_negative_derivative(max("bytes_recv"), 1s) AS "received", non_negative_derivative(max("bytes_sent"), 1s) AS "sent" FROM "telegraf"."autogen"."net" WHERE time > :dashboardTime: AND "host" = :host: GROUP BY time(:interval:) FILL(null)`
- cpu / mem per process: based on procstat